### PR TITLE
57 filter component

### LINKED ISSUE #57: Filter Component
- [x] get categories form /api/categories route

- [x] render catergories in radio group 
- [x] add search component and handle change function
- [x] hnandle isAvailable change and add its component

### PR DESCRIPTION
**Pull request by farah2003**
relates #57
closes #57 
- [x] get categories form /api/categories route

- [x] render catergories in radio group 
- [x] add search component and handle change function
- [x] hnandle isAvailable change and add its component
* Signed in only:	`false`
* Admin only:	`false`
image4

image2

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -1,40 +0,0 @@
-name: Indonate
```
Comment (Mohammed-Hamada) [inline .github/workflows/node.js.yml:1]: Why did you remove this file?

File: client/src/components/filterCampaines/index.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  message,
+  Radio,
+  Input,
+  Segmented,
+
+} from 'antd';
+import './style.less';
+
+const { Search } = Input;
```
Comment (Mohammed-Hamada) [inline client/src/components/filterCampaines/index.jsx:13]: you can import it directly, without declare it again
```js
Input: { Search }
```

File: client/src/components/filterCampaines/index.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      try {
+        const { data: { data: { categories: categoriesFromDB } } } = await axios('/api/categories');
+        const categoriesNameArr = ['List All'];
+        categoriesFromDB.map((item) => categoriesNameArr.push(item.name));
+        setCategories(categoriesNameArr);
+      } catch ({ response: { data: { message: errorMessage } } }) {
+        message.info(errorMessage);
+      }
+    };
+    fetchData();
```
Comment (Mohammed-Hamada) [inline client/src/components/filterCampaines/index.jsx:28]: What this line do?

File: client/src/components/filterCampaines/index.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }, []);
+  const handleChangeCategories = ({ target: { value } }) => {
+    setCategory(value);
+  };
+  const HandleSearchChange = ({ target: { value } }) => {
```
Comment (Mohammed-Hamada) [inline client/src/components/filterCampaines/index.jsx:33]: Unify the naming of variables.

**Review by Mohammed-Hamada — CHANGES_REQUESTED** [review]

File: client/src/components/filterCampaines/index.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  message,
+  Radio,
+  Input,
+  Segmented,
+
+} from 'antd';
+import './style.less';
+
+const { Search } = Input;
+function Compaigns({ setCategory, setAvailable, setSearch }) {
```
Comment (HaniOlwan) [inline client/src/components/filterCampaines/index.jsx:14]: Campaigns not Compaigns

Comment (HaniOlwan) [reply client/src/components/filterCampaines/index.jsx:14]: also change the name of the file to  filterCampagins

File: client/src/components/filterCampaines/index.jsx (hunk 3 above)
Comment (HaniOlwan) [reply client/src/components/filterCampaines/index.jsx:28]: it fetches all the categories so we display them in front

File: client/src/components/filterCampaines/index.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  const HandleSearchChange = ({ target: { value } }) => {
+    setSearch(value);
+  };
+  const handleAailableChange = (e) => {
+    setAvailable(e);
+  };
+  return (
+    <div className="fliter-section">
+      <div>
+        <Radio.Group onChange={handleChangeCategories} defaultValue="List All" buttonStyle="solid">
```
Comment (HaniOlwan) [inline client/src/components/filterCampaines/index.jsx:42]: destructure group

File: client/src/components/filterCampaigns/index.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      try {
+        const { data: { data: { categories: categoriesFromDB } } } = await axios('/api/categories');
+        const categoriesNameArr = ['List All'];
+        categoriesFromDB.map((item) => categoriesNameArr.push(item.name));
+        setCategories(categoriesNameArr);
+      } catch ({ response: { data: { message: errorMessage } } }) {
+        message.info(errorMessage);
+      }
+    };
+    fetchData();
```
Comment (Raghad-Mezied) [inline client/src/components/filterCampaigns/index.jsx:29]: You need to cleanup after the request.

File: client/src/components/filterCampaigns/index.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+function Campaigns({ setCategory, setAvailable, setSearch }) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { data: { data: { categories: categoriesFromDB } } } = await axios('/api/categories');
+        const categoriesNameArr = ['List All'];
+        categoriesFromDB.map((item) => categoriesNameArr.push(item.name));
+        setCategories(categoriesNameArr);
```
Comment (Raghad-Mezied) [inline client/src/components/filterCampaigns/index.jsx:24]: Cant you just set the category like this
```js
setCategories(['List All', ...categoriesFromDb])
```
And you can just rename the categories state categoriesList so you don't rename in line 21

File: client/src/components/filterCampaigns/index.jsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  };
+  const handleAailableChange = (e) => {
+    setAvailable(e);
+  };
+  return (
+    <div className="fliter-section">
+      <div>
+        <Group onChange={handleCategoryChange} defaultValue="List All" buttonStyle="solid">
+          {categories.map((item) => <Radio.Button value={item}>{item}</Radio.Button>)}
+        </Group>
```
Comment (Raghad-Mezied) [inline client/src/components/filterCampaigns/index.jsx:45]: Destructure the Button here, and don't forget to add a key when mapping.

File: client/src/components/filterCampaigns/index.jsx [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      </div>
+
+      <Segmented options={['Avilable', 'Not Avilable']} onChange={handleAailableChange} />
+      <Search placeholder="input search text" onChange={handleSearchChange} style={{ width: '300px' }} />
```
Comment (Raghad-Mezied) [inline client/src/components/filterCampaigns/index.jsx:49]: Move the style to your CSS files

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Comment by Raghad-Mezied** [comment]
Add in the description of the pull request a screenshot for your final component.
